osh/cmd_parse.py

```diff
@@ -1175,7 +1183,7 @@ def _ParseForExprLoop(self, for_kw):
     self._Peek()
     if self.c_id == Id.Op_Semi:
       self._Next()
-      self._NewlineOk()
+      self._NewlineOkLegacy()
```

Comment:
Since this change is likely to be big, I think it's better to do the reverse change, e.g. add _NewlineOkNew(), and then flip the names in a subsequent change